giraph-core/src/main/java/org/apache/giraph/counters/CustomCounters.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.giraph.counters;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Custom counters created by the client or app
+ * Stores the names of the counters, and the final aggregated values
+ */
+public class CustomCounters {
+
+
+
+  /** Unique counter groups and names populated during execution of the job */
+  private static Set<CustomCounter> COUNTER_NAMES = new HashSet<>();
+
+  /** Aggregated counter values updated by master */
+  private final List<CustomCounter> counterList;
```

Comment:
Since the basic operation you're doing on the list is search and update the value, maybe make this a Set for faster lookups?

Reply:
I cannot use a set because I need to be able to get the element, update its contents and insert it back into the set. But, hash set does not support a method to fetch an element from the set.